include/xtensor-io/xchunk_store_manager.hpp

```diff
@@ -138,6 +137,16 @@ namespace xt
         IP m_index_path;
     };
 
+    template <class EC, class IP, class EXT = empty_extension, class S>
+    xchunked_array<xchunk_store_manager<EC, IP>, EXT> chunked_array(S&& shape, S&& chunk_shape, const std::string& path);
```

Comment:
Maybe you can add a `pool size` argument ? I think we had it in the `xchunked_array` constructors before we refactored them.

You may also add a function for building the array from an expression.

Reply:
> Maybe you can add a `pool size` argument ? I think we had it in the `xchunked_array` constructors before we refactored them.

:+1: 

> You may also add a function for building the array from an expression.

Isn't it better in this case to just use the above factory function, and assign the expression to it? The problem otherwise is that the data type information in the expression is not enough to deduce a chunk container. Of course we could default to e.g. an `xfile_array` with a disk IO handler. But still, the user has to then set a path, so it's not usable by default.
I think building a chunked array from an expression should default to an in-memory chunked array, which we already have implemented.

Reply:
> Isn't it better in this case to just use the above factory function, and assign the expression to it?

The issue with that approach is that you go through unecessary temporary and reshaping, while a constructor accepting an `xepxression` can avoid that.

> Of course we could default to e.g. an xfile_array with a disk IO handler. But still, the user has to then set a path, so it's not usable by default.

You can add arguments to the factory method (this is what is done for the in-memory chunked array). Constructors accepting an `xexpression` only are not possible for `xchunked_array`.

Reply:
Forget what I said, we are passing the path already.